instrumentation.go

```diff
@@ -18,11 +18,51 @@ import (
 	"golang.org/x/tools/cover"
 )
 
-// InstrumentAndLoadCollection adds instrumentation instructions to all programs contained within the given collection.
-// This "instrumentation" is nothing more than incrementing a 16-bit number within a map value, at an index unique
-// to the location within the program(Block ID/index). After updating the program, it is loaded into the kernel, the
-// loaded collection and a list of program blocks is returned. The index of the returned program blocks matches the
-// index of blocks in the coverage map.
+// InstrumentAndLoadCollection will instrument the given collection spec and proceed to load it using the provided
+// options. Please refer to `InstrumentCollection` for more information about the instrumentation process.
+func InstrumentAndLoadCollection(
+	coll *ebpf.CollectionSpec,
+	opts ebpf.CollectionOptions,
+	logWriter io.Writer,
+) (*ebpf.Collection, []*BasicBlock, error) {
+	blockList, err := InstrumentCollection(coll, logWriter)
+	if err != nil {
+		return nil, nil, fmt.Errorf("InstrumentCollection: %w", err)
+	}
+
+	if logWriter != nil {
+		// Verbose
+		opts.Programs.LogLevel = 2
+	}
+
+	loadedColl, err := ebpf.NewCollectionWithOptions(coll, opts)
+
+	if logWriter != nil {
+		fmt.Fprintln(logWriter, "=== Instrumented verifier logs ===")
+		if loadedColl != nil {
+			for name, prog := range loadedColl.Programs {
+				fmt.Fprintln(logWriter, "---", name, "---")
+				fmt.Fprintln(logWriter, prog.VerifierLog)
+			}
+		}
+		if err != nil {
+			var vErr *ebpf.VerifierError
+			if errors.As(err, &vErr) {
+				fmt.Fprintln(logWriter, strings.Join(vErr.Log, "\n"))
```

Comment:
Sorry I didn't catch this earlier; is the goal to retrieve the full (unsummarized) verifier log? That can be done with:

```suggestion
				fmt.Fprintf(logWriter, "%+v\n", vErr)
```